sunpy/coordinates/transformations.py

```diff
@@ -197,52 +213,64 @@ def hpc_to_hpc(heliopcoord, heliopframe):
     return hpc
 
 
-# Make a transformation graph for the documentation, borrowed lovingly from
-# Astropy.
+def _make_rotation_matrix(start_representation, end_representation):
+    """
+    Return the matrix for the direct rotation from one representation to a second representation.
+    The representations need not be normalized first.
+    """
+    A = start_representation.to_cartesian()
+    B = end_representation.to_cartesian()
+    rotation_axis = A.cross(B)
+    rotation_angle = -np.arccos(A.dot(B) / (A.norm() * B.norm()))  # negation is required
+
+    # This line works around some input/output quirks of Astropy's rotation_matrix()
+    matrix = np.array(rotation_matrix(rotation_angle, rotation_axis.xyz.value.tolist()))
+    return matrix
+
 
+# The Sun's north pole is oriented RA=286.13 deg, dec=63.87 deg in ICRS, and thus HCRS as well
+# (See Archinal et al. 2011,
+#   "Report of the IAU Working Group on Cartographic Coordinates and Rotational Elements: 2009")
+# The orientation of the north pole in ICRS/HCRS is assumed to be constant in time
+_solar_north_pole_hcrs = UnitSphericalRepresentation(lon=286.13*u.deg, lat=63.87*u.deg)
 
-def _make_transform_graph_docs():
+
+# Calculate the rotation matrix to de-tilt the Sun's rotation axis to be parallel to the Z axis
+_sun_detilt_matrix = _make_rotation_matrix(_solar_north_pole_hcrs,
+                                           CartesianRepresentation(0, 0, 1))
+
+
+@frame_transform_graph.transform(DynamicMatrixTransform, HCRS, HeliographicStonyhurst)
+def hcrs_to_hgs(hcrscoord, hgsframe):
     """
-    Generates a string for use with the coordinate package's docstring
-    to show the available transforms and coordinate systems
+    Convert from HCRS to Heliographic Stonyhurst.
     """
-    import inspect
-    from textwrap import dedent
-    from sunpy.extern import six
-    from astropy.coordinates.baseframe import (BaseCoordinateFrame,
-                                               frame_transform_graph)
+    if hgsframe.obstime is None:
+        raise ValueError("To perform this transformation the coordinate"
+                         " Frame needs a obstime Attribute")
 
-    import copy
-    f = copy.deepcopy(frame_transform_graph)
-    for f1 in frame_transform_graph._graph.keys():
-        if 'sunpy' not in str(f1):
-            del f._graph[f1]
-        else:
-            for f2 in frame_transform_graph._graph[f1].keys():
-                if 'sunpy' not in str(f2):
-                    del f._graph[f1][f2]
+    # Determine the Sun-Earth vector in this de-tilt frame
+    sun_pos_icrs = get_body_barycentric('sun', hgsframe.obstime)
+    earth_pos_icrs = get_body_barycentric('earth', hgsframe.obstime)
+    sun_earth_hcrs = earth_pos_icrs - sun_pos_icrs
```

Comment:
It might be a good idea to leave a comment here specifically pointing out that this works because HCRS and ICRS have the same orientation.  otherwise it looks very confusing that you use the ICRS positions to compute an HCRS position. (I could imagine a well-meaning person reading this and thinking it was a mistake...)